File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;

public class Main {
    static int[] visit = new int[1000001];

    public static void BFS(int n){
        Queue<Integer> q = new LinkedList<>();
        q.add(n);
        visit[n] = 1;

        while(!q.isEmpty()){
            n = q.poll();

            if(n < 1) break;

            // 3으로 나누어 떨어지고 아직 방문 X
            if(n%3 == 0 && visit[n/3] == 0){
                q.add(n/3);
                visit[n/3] = visit[n] + 1;
            }
            // 2로 나누어 떨어지고 아직 방문 X
            if(n%2 == 0 && visit[n/2] == 0){
                q.add(n/2);
                visit[n/2] = visit[n] + 1;
            }
            // -1 아직 방문 X
            if(visit[n-1] == 0){
                q.add(n-1);
                visit[n-1] = visit[n] + 1;
            }
        }
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        BFS(N);
        System.out.println(visit[1] - 1);
    }
}

File: 백준/Silver/1463. 1로 만들기/1로 만들기.cc
#include <iostream>
#include <queue>

using namespace std;

int visit[1000001];

void BFS(int n){
    queue<int> q;
    q.push(n);
    visit[n] = 1;

    while(!q.empty()){
        int cur = q.front();
        q.pop();

        if(cur < 1) break;

        // 3으로 나누어 떨어지고 아직 방문 X
        if(cur%3 == 0 && visit[cur/3] == 0){
            q.push(cur/3);
            visit[cur/3] = visit[cur] + 1;
        }
        // 2로 나누어 떨어지고 아직 방문 X
        if(cur%2 == 0 && visit[cur/2] == 0){
            q.push(cur/2);
            visit[cur/2] = visit[cur] + 1;
        }
        // -1 아직 방문 X
        if(visit[cur-1] == 0){
            q.push(cur-1);
            visit[cur-1] = visit[cur] + 1;
        }
    }
}

int main(){
    int N;
    cin >> N;
    BFS(N);
    cout << visit[1] - 1;

    return 0;
}